build.gradle.kts

```diff
@@ -0,0 +1,199 @@
+            r.lineSequence().last()
+        }
+
+val branch = gitBranch()
+logger.info("On branch $branch")
+
+group = "be.bluexin"
+version = "${if (branch == "master") "" else "$branch-"}${prop("version_number")}.${prop("build_number")}"
+
+repositories {
+    jcenter()
+}
+
+configurations {
+    arrayOf(
+            "shade",
+            "shadeInPlace"
+    ).forEach {
+        create(it) {
+            get("compileOnly").extendsFrom(this)
+            get("testCompileOnly").extendsFrom(this)
+        }
+    }
+}
+
+dependencies {
+    api(kotlin("stdlib-jdk8"))
+    api("org.jetbrains.kotlin:kotlin-stdlib-jdk8")
+    api("org.jetbrains.kotlinx:kotlinx-coroutines-core:0.19.3")
+    api("org.jetbrains.kotlinx:kotlinx-coroutines-jdk8:0.19.3")
+    api("org.jetbrains.kotlinx:kotlinx-coroutines-nio:0.19.3")
+
+    implementation("org.slf4j:slf4j-api:1.7.25")
+    implementation("io.github.microutils:kotlin-logging:1.6.10")
+    implementation("org.slf4j:slf4j-simple:1.7.25")
+
+    "shade"("net.java.dev.jna:jna:4.5.0")
+    "shadeInPlace"(files("libs"))
+    "shadeInPlace"(files("libsExt") {
+        builtBy("expandDiscordRPC")
+    })
+}
+
+tasks.withType<KotlinCompile> {
+    kotlinOptions {
+        jvmTarget = "1.8"
+        languageVersion = "1.2"
+        apiVersion = "1.2"
+    }
+}
+
+kotlin {
+    experimental.coroutines = Coroutines.ENABLE
+}
+
+val sourceJar by tasks.registering(Jar::class) {
+    from(sourceSets["main"].allSource)
+    from(sourceSets["test"].allSource)
+    classifier = "sources"
+}
+
+tasks.withType<Jar> {
+    for (dep in configurations["shade"]) {
+        from(project.zipTree(dep)) {
+            exclude("META-INF", "META-INF/**")
+        }
+    }
+    for (dep in configurations["shadeInPlace"]) {
+        from(dep)
+    }
+}
+
+publishing {
+    publications {
+        register("drpc4k", MavenPublication::class.java) {
+            from(components["java"])
+            artifact(sourceJar.get())
+        }
+    }
+
+    repositories {
+        val remote = System.getenv("REPO_PWD") != null
+        maven {
+            val remoteURL = "https://maven.bluexin.be/repository/" + (if ((version as String).contains("SNAPSHOT")) "snapshots" else "releases")
+            val localURL = "file://$buildDir/repo"
+            url = uri(if (remote) remoteURL else localURL)
+            if (remote) {
+                credentials(PasswordCredentials::class.java) {
+                    username = "CI"
+                    password = System.getenv("REPO_PWD")
+                }
+            }
+        }
+    }
+}
+
+fun downloadOne(url: String, filename: String) {
+    file("tmp").mkdir()
+    val connection = uri(url).toURL().openConnection()
+    connection.inputStream.use {
+        file("tmp/$filename").outputStream().use { output ->
+            var r = it.read()
+            while (r >= 0) {
+                output.write(r)
+                r = it.read()
+            }
+        }
+    }
```

Comment:
I think Kotlin has a built-in function to redirect an inputStream to an outputStream

Reply:
I planned to switch to using a Gradle plugin to handle downloads better, as well as keep them cached (instead of downloading pretty much whenever a Gradle task is run)